vue/components/ui/molecules/upload-area/upload-area.vue

```diff
@@ -2,45 +2,37 @@
     <div
         class="upload-area"
         v-bind:class="classes"
-        v-on:dragover.prevent="event => onDragOver(event)"
-        v-on:drop.prevent="event => onDrop(event)"
-        v-on:dragenter="event => onDragEnter(event)"
-        v-on:dragleave="event => onDragLeave(event)"
+        v-on:dragover.prevent="onDragOver"
+        v-on:drop.prevent="onDrop"
+        v-on:dragenter="onDragEnter"
+        v-on:dragleave="onDragLeave"
     >
-        <input
-            type="file"
-            hidden
-            v-bind:multiple="multiple"
-            v-bind:accept="accept"
-            ref="filesInput"
-            v-on:change="onFilesInputChange"
-        />
-        <slot v-bind:open-modal="openModal">
-            <div class="upload-area-container">
-                <transition name="fade-in" mode="out-in">
-                    <div class="description" v-bind:key="dragging">
-                        {{ descriptionText }}
-                    </div>
-                </transition>
-                <button-icon v-bind:icon="icon" v-bind:size="110" />
-                <button-color
-                    class="button-upload"
-                    v-bind:text="'Upload File'"
-                    v-bind:icon="'cloud-upload'"
-                    v-bind:alignment="'center'"
-                    v-bind:disabled="disabled"
-                    v-on:click="onUploadButtonClick"
-                />
-            </div>
+        <slot>
+            <transition name="fade-in" mode="out-in">
+                <div class="description" v-bind:key="dragging">
+                    {{ descriptionText }}
+                </div>
+            </transition>
+            <icon v-bind:icon="icon" v-bind:size="110" v-bind:color="'gray'" />
         </slot>
+        <upload-button

```

Comment:
Couldn't this simply be a substitute of the existing `button-color`, leaving the structure as it was? This change has some issues:
- Being outside the slot means it can't be overridden
- The former point plus the fact that the HTML changed is a breaking change of sorts

Reply:
The problem is that previously we had this outside the slot:
```
        <input
            type="file"
            hidden
            v-bind:multiple="multiple"
            v-bind:accept="accept"
            ref="filesInput"
            v-on:change="onFilesInputChange"
        />
 ```
 Now that native input is inside `upload-button` we had to move it ouside the slot to replace the native input.
The native input has to somehow be outside the slot for the upload mixin logic to run using it.

An alternative could be moving back `<input>` in this component and toggling it between native or the one from upload-button when we have slots. I didn't do it because it went a little bit against the philosohpy of separating and re-using the components.

Visually, the upload-button becomes invisible with `display: none` when we the component receives slots so there's no visual impact in this change.